### PR TITLE
GPU: HILO8/RG8

### PR DESCRIPTION
**Pull request by ILOVEPIE**
New PR for HILO8/RG8 see PR #773
Has been tested with this: https://github.com/yuriks/3DS_gpu_tests/tree/hilo8-test

### CODE REVIEW

File: src/video_core/color.h [hunk 1]
```diff
@@ -70,6 +70,17 @@ inline const Math::Vec4<u8> DecodeRGB8(const u8* bytes) {
 }
 
 /**
+ * Decode a color stored in RG8 (HILO8) format
+ * @attention The technically correct name for this format is HILO8 due to it being big-endian regardless of archetecture, because the 3ds is little-endian we have used the name RG8 to prevent confusion.
```
Comment (yuriks) [inline src/video_core/color.h]: Hm... I don't think any of this matters? It follows the same endianess conventions as the rest of the non-packed color formats.

File: src/video_core/color.h [hunk 2]
```diff
@@ -70,6 +70,17 @@ inline const Math::Vec4<u8> DecodeRGB8(const u8* bytes) {
 }
 
 /**
+ * Decode a color stored in RG8 (HILO8) format
```
Comment (yuriks) [inline src/video_core/color.h]: "(also known as HILO8)"

File: src/video_core/color.h [hunk 3]
```diff
@@ -153,6 +164,17 @@ inline void EncodeRGB8(const Math::Vec4<u8>& color, u8* bytes) {
...
 /**
+ * Encode a color as RG8 (HILO8) format
+ * @attention The technically correct name for this format is HILO8 due to it being big-endian regardless of archetecture, because the 3ds is little-endian we have used the name RG8 to prevent confusion.
+ * @remark RG8 is not intended to store colors, the word color here is used just for consistency, RG8 is a format intended for normal maps.
+ * @param color Source color to encode
+ * @param bytes Destination pointer to store encoded color
+ */
+inline void EncodeRG8(const Math::Vec4<u8>& color, u8* bytes) {
+    bytes[0] = color.g();
+    bytes[1] = color.r();
```
Comment (yuriks) [inline src/video_core/color.h]: Swap these two lines to stay consistent with the ordering used in the rest of the functions.

File: src/video_core/pica.h [hunk 4]
```diff
@@ -152,7 +152,7 @@ struct Regs {
         RGB565       =  3,
         RGBA4        =  4,
         IA8          =  5,
-
+        RG8          =  6,  /// @attention The technically correct name for this format is HILO8 due to it being big-endian regardless of archetecture, because the 3ds is little-endian we have used the name RG8 to prevent confusion.
```
Comment (yuriks) [inline src/video_core/pica.h]: This comment about endiness doesn't even make sense to me. It's just a regular texture format.

Comment (ILOVEPIE) [inline src/video_core/pica.h]: It has a fixed endianness across platforms

Comment (neobrain) [inline src/video_core/pica.h]: Agree that there is no point in this comment. For all purposes relevant to emulation, we can refer to it as just another texture format. The fact that it also matches the specification for HILO8 is nice to know to research about the applications of that format, but it's far from crucial and not really informative in any other way.

File: src/video_core/color.h (hunk 1 above)
Comment (neobrain) [inline src/video_core/color.h]: +1, also please don't duplicate explanations like this.

**Comment by neobrain** [comment]
Shouldn't the texture viewing widget be updated for this too? (At least a new entry should be added to the texture format selection.)

Feel free to ask me on IRC if you need more info about this.

**Comment by neobrain** [comment]
Please include a reference to the hardware test in the PR description.

**Comment by bunnei** [comment]
@ILOVEPIE can we wrap this up please?

**Comment by ILOVEPIE** [comment]
yeah sorry, been busy with some other stuff

**Comment by yuriks** [comment]
I'm going to rebase/fix and PR this soon.

**Comment by ILOVEPIE** [comment]
Sorry i've been busy with family issues, I'll get this rebased either today
or tommorow.

**Comment by bunnei** [comment]
@ILOVEPIE ping again on this. If you don't have the time, please let us know and one of the other contributors will complete this change - thanks.

**Comment by yuriks** [comment]
I took this over on #1034

**Pull request closed without merging**